### DISCUSSION TITLE
Solar divert?

### DISCUSSION THREAD

**Discussion by scrauny** [raomin/ESPAltherma#164] [Ideas]
This project looks awesome, congratulations.

I was wondering if it is possible to instruct the heat pump to heat hot water using a power limit in Watts?

i.e. Home Assistant could tell the ESPAltherma to tell the Daikin to heat hot water with a limit of X Watts, which it would then continuously update to reflect the available excess solar generation?

The Daikin LAN controller brp069a61 has an input for a pulse based meter so it seems within the realms of possibility! But the LAN controller is expensive, difficult to get, and doesn't have WiFi.

Many thanks!

**Comment by raomin** [comment]
Hi @scrauny, thanks for the feedback.
What you describe is what is referred to has 'Smart Grid integration'.

Daikin has a module for setting 3 levels of power use. It's documented here https://github.com/raomin/ESPAltherma#step-4-optional---smart-grid-features and the feature was added by @andreapier.
Just add 2 relays and you'll be good to go.

**Reply by scrauny** [reply]
Hi,

Thanks for the reply.

Yes, in Smart Grid mode SG2 ("Recommended ON") it is possible to restrict the Heat Pump to either a static or dynamic power limitation. With the Daikin adapter (brp069a61) you can connect a pulse meter, and then somehow it tells the Heat Pump what the power limit is. Daikin blurb and link below.

Their adapter uses ports X5M/1+2 (seems a different name for P1/P2) on the indoor unit, so perhaps it is a different protocol. It'd certainly be a nice feature. We can emulate the pulse meter with the ESP reading some value from Home Assistant.

Thanks,
Shaun

https://www.daikin.eu/content/dam/document-library/Installer-reference-guide/ctrl/BRP069A61,%20BRP069A62_4PEN464229-1_2017_08_Installer%20reference%20guide_English.pdf

Dynamic power limitation
To enable dynamic power limitation, the system requires an
electrical meter. In this case, the power limitation is auto-adaptive,
and dynamically performed based on the power injection into the
grid, measured by the electrical meter.

**Reply by bmesuere** [reply]
I'm using the static power limitation for smart grid, and for me it doesn't work. I have a Shelly EM measuring power draw and the unit always uses >2kW while I have a limit set at 1.5kW.

**Reply by javimurcia** [reply]
> I'm using the static power limitation for smart grid, and for me it doesn't work. I have a Shelly EM measuring power draw and the unit always uses >2kW while I have a limit set at 1.5kW.

I think there is a minimum power budget you have to allocate to the heat pump.

**Reply by javimurcia** [reply]
> Hi,
> 
> Thanks for the reply.
> 
> Yes, in Smart Grid mode SG2 ("Recommended ON") it is possible to restrict the Heat Pump to either a static or dynamic power limitation. With the Daikin adapter (brp069a61) you can connect a pulse meter, and then somehow it tells the Heat Pump what the power limit is. Daikin blurb and link below.
> 
> Their adapter uses ports X5M/1+2 (seems a different name for P1/P2) on the indoor unit, so perhaps it is a different protocol. It'd certainly be a nice feature. We can emulate the pulse meter with the ESP reading some value from Home Assistant.
> 
> Thanks, Shaun
> 
> https://www.daikin.eu/content/dam/document-library/Installer-reference-guide/ctrl/BRP069A61,%20BRP069A62_4PEN464229-1_2017_08_Installer%20reference%20guide_English.pdf
> 
> Dynamic power limitation To enable dynamic power limitation, the system requires an electrical meter. In this case, the power limitation is auto-adaptive, and dynamically performed based on the power injection into the grid, measured by the electrical meter.

Yes, the smart wifi adapter communicates with the  heat pump via the P1P2 interface. Newer units have the smart grid terminals, and the pulse meter input built in.

I think emulating a P1P2 device it's outside the scope of this project, but maybe generating said pulse signal might be doable

**Reply by xalex75** [reply]
> I'm using the static power limitation for smart grid, and for me it doesn't work. I have a Shelly EM measuring power draw and the unit always uses >2kW while I have a limit set at 1.5kW.

Same here. I'm using smart grid, and that's ok. But with any power limit set on the machine panel, ACS always uses 3/3.4kW
